
# Exercício 1
## 1. Crie 3 variáveis:

nomePrimeiro, nomeMeio, nomeUltimo

Atribua um valor para cada uma delas e imprima a seguinte frase para o usuário:

"Gabriel Gonçalves de Briro, seja muito bem vindo a DNC"

Não precisa usar input( )

In [28]:
nomePrimeiro= 'Gabriel '
nomeMeio= 'Gonçalves de '
nomeUltimo= 'Brito'

print(nomePrimeiro + nomeMeio + nomeUltimo + ', seja bem vindo a DNC')

Gabriel Gonçalves de Brito, seja bem vindo a DNC


## 2. Agora peça para o usuário digitar as 3 variáveis antes de mostrar a mesma frase da parte 1.

Resultado esperado

In [29]:
nomePrimeiro=input('Qual o seu primeiro nome?\n')
nomeMeio=input('Qual o seu nome do meio?\n')
nomeUltimo=input('Qual o seu ultimo nome?\n')

print(nomePrimeiro, nomeMeio, nomeUltimo + ', seja muito bem vindoa DNC')

Gabriel Gonçalves de Brito, seja muito bem vindoa DNC


# Exercício 2
#### 1. Vamos criar nosso primeiro script 💪, que:

Armazene em uma variavel o nome do cliente, a variável precisa se chamar cliente_nome e o nome do cliente você pode usar sua criatividade.

Crie uma variável que armazene a idade do cliente, a variável precisa se chamar cliente_idade.

Exiba na tela as 2 variáveis e apenas uma variável por linha. Ex: "O nome do Cliente é: Nathan Pereira".

In [24]:
cliente_nome='Gabriel Gonçalves de brito'
cliente_idade= 25

print('Nome do cliente:', cliente_nome)
print('A idade do cliente:', cliente_idade)

Nome do cliente: Gabriel Gonçalves de brito
A idade do cliente: 25


#### 2. Agora que você já criou seu primeiro script vamos aumentar a dificuldade!
Agora é o cliente que vai colocar as informações! Crie uma variável que receba como input o nome e a idade do cliente!
Para ajudar o usuário exiba, essas variáveis na tela.

In [3]:
cliente= input('Qual seu nome?\n')
idade= input('Qual a sua idade?\n')

print('O nome do clinte é:', cliente)
print('A idade do cliente é:', idade)

O nome do clinte é: Gabriel Gonçalves de Brito
A idade do cliente é: 25


#### 3. Vamos brincar com mais variáveis agora?!
Crie uma variável do tipo FLOAT que vai armazenar o valor do produto, que receberá o valor de 25.30!
Crie uma variável do tipo BOOL que vai armazenar se ele é ou não um cliente premium, recebendo o valor de True.
Agora exiba essas variáveis na tela.
Exiba também o tipo dessas variáveis.

In [20]:
produto_valor=25.30
cliente_premium=True

print('O valor do produto é:', produto_valor)
print ('O tipo da variável produto-valor é:', type( produto_valor))
print('O cliente é premium?', cliente_premium)
print('O tipo da variável cliemte_premium é:', type(cliente_premium))
print()

print('o valor do produto é: ,produto_valor', produto_valor, type (produto_valor))
print('o cliente é premium?', cliente_premium, type(cliente_premium))

O valor do produto é: 25.3
O tipo da variável produto-valor é: <class 'float'>
O cliente é premium? True
O tipo da variável cliemte_premium é: <class 'bool'>

o valor do produto é: ,produto_valor 25.3 <class 'float'>
o cliente é premium? True <class 'bool'>


### tipos de dados

*str:* cadeia de caracteres= ']~´[@', '7', 'Gabriel'

*int:* inteiro= 5, 10, 25

*float:* real= 5.3, 25.3

*bool:* booleanos= true, false


### funções

**print()=** imprimr uma frase

**type()=** verifica o tipo de dado de uma variável

**input()=** pede pro usuário fornecer um valor (str)

# plote as variáveis a baixo:
## a = 245
## b = 2.835895

In [41]:
a = 245
b = 2.835895

print('| {0:10} | e | {1:10.3f} |'.format(a, b))

|        245 | e |      2.836 |


### Fui numa hamburgueria e comprei 2 lanches de 10,90 cada, 3 refris de 4,50 cada e 1 batata por 8,45.
### O pagamento foi no dinheiro com notas de 10 reais e 5 reais. perguntas
1. Qual foi o valor total da conta?
2. Quantas notas de 10 reais eu usei?
3. Quantas notas de 5 reais eu usei?
4. Quanto voltou de troco para mim em notas?
5. Qunato voltou de troco para mim em moedas?

In [38]:
qunatidadeLanche=2
quantidadeRefri=3
quantidadebatata=1
valorlanche=10.9
valorRefri=4.5
valorBatata=8.45

valorTotal= qunatidadeLanche * valorlanche + quantidadeRefri * valorRefri + quantidadebatata * valorBatata

print('O valor da conta foi de: R$ {}'.format(valorTotal))

notas10=valorTotal // 10

print('Você tera que utilizar: {} notas de R$ 10,00'.format(int(notas10)))

valorFalta = valorTotal - notas10 * 10 
notas5 = valorFalta // 5 + 1
valorFalta= valorFalta - notas5

print('Você utilizara {} notas de R$ 5,00'. format (int(notas5)))

valorFalta = valorFalta - notas5
notas2= valorFalta // 2


print('Eu recebi {} notas de R$ 2,00'.format(int(notas2)))

valorMoeda= valorFalta - 0

print('Vou receber R$ {} em moedas'.format(valorMoeda))


O valor da conta foi de: R$ 43.75
Você tera que utilizar: 4 notas de R$ 10,00
Você utilizara 1 notas de R$ 5,00
Eu recebi 0 notas de R$ 2,00
Vou receber R$ 1.75 em moedas


# Exercícios
## Exercício 1:
#### Vamos testar os comportamentos dos diferentes operadores aritméticos. Para isso vamos montar uma calculadora, eai? Bora, vamos nessa?

Criar variáveis n1 e n2 que receberão inputs do tipo INT do usuário. (não se esqueça de tratar o que está sendo recebido rsrs) Exibir o resultado da Adição dessas 2 variáveis. Exibir o resultado da Subtração dessas variáveis. Exibir o resultado da Multiplicação dessas variáveis. Exibir o resultado da Divisão dessas variáveis. Exibir o resultado do Módulo (Resto da divisão) entre essas variáveis. Exibir o resultado de uma variável Exponenciando a outra. [ ]

In [2]:
n1=int(input("igite um valor inteiro"))
n2=int(input('informe mais um valor inteiro'))

print ('n1:',n1,
'\nn2:',n2)
print (n1+n2)
print (n1-n2)
print (n1*n2)
print (n1/n2)
print (n1//n2)
print (n1%n2)

n1: 7 
n2: 5
12
2
35
1.4
1
2


## Exercício 2:
Faça um programa que o usuário digita um valor inteiro e nós mostramos toda a tabuada daquele número.

In [42]:
contador = 0
valor=int(input('informe um valor inteiro\n'))
while True:
    if contador < 10:
        contador = contador + 1
        if contador :
            print(('{} * {} = {}'.format(valor, contador, valor*contador)))
    else:
        break
print("chegamos ao fim, até a proxima")

18 * 1 = 18
18 * 2 = 36
18 * 3 = 54
18 * 4 = 72
18 * 5 = 90
18 * 6 = 108
18 * 7 = 126
18 * 8 = 144
18 * 9 = 162
18 * 10 = 180
chegamos ao fim, até a proxima


## Exercício 3:
Faça um programa no qual o usuário digite a distância percorrida por um carro e a qtd de litros de combustível gasto.
Calcule e mostre para o usuário quantos km/l o seu carro faz.

In [45]:
quilometros=int(input('Quantos km você percorreu com seu carro? \n'))
litrosConbustivel=int(input('Quantos litros de conbustível foi gastos?\n'))

km_L= quilometros / litrosConbustivel

print('Seu carro faz {:.2f}km/L'.format(km_L))

Seu carro faz 16.67km/L


## Exercício
#### dado o peço do lanche = 10,90 e da batata de 5,40. calcule quando vai sair a conta do usuário. o usoário deve inserir a quantidade de lanche que ele quer e a quantidade de batatas. finalmentr peça para o usuário falar quanto dinheiro ele tem na carteira e verifique se ele tem vai consegir pagar a cconta.

In [10]:
lancheV=10.9
batataV=5.4
lancheQ=float(input('Quantos lanches você quer?\n'))
batataQ=float(input('Quantas batata voce quer?\n'))
dinheiro=float(input('Quanto de dinehiro você tem?\n'))

total= lancheV*lancheQ + batataV * batataQ
if dinheiro >= total:
    print('você tem dinheiro suficiente par apagar')
else:
    print('você não consegue pagar a conta')

você tem dinheiro suficiente par apagar


## Exercício 2
#### 1. faça um programa que pede para um usuário siditar 2 números e:

se num1 >= num2 estão multiplique o num2 por 5.

caso contratio, multiplique o num1 por 5

In [29]:
num1= float(input('digite um número: '))
num2= float(input('digite outro número: '))

if num1>=num2:
    print('num1 é maior ou igual a num2')
    num2=num2*5

else:
    print('num2 é maior ou igual a num1 ')
    num1 = num1*5

print('num1 {}, num2 {}'.format(num1, num2))

num2 é maior ou igual a num1 
num1 5.0, num2 3.0


# exercício 3

Faça um programa que verifique se o usuário digitpu um valor numérico ou não.

1. caso verdadeiro, inprima: "Esse input é um número"
2. caso falso, imprima: "Esse input não é um número, tente novamente" 

In [6]:
teste=input('digite algo')

numero=teste.isnumeric()
texto = teste.isalpha()

if numero == True :
    print('Esse input e um número')
elif texto == True :
    print('Você digitou texto.\n'
    'Tente novamente,. digite apenas numeros')
else:
    print('Esse input é texto com número\n'
    'Preciso apenas de números, tente novamente!')

# CONDICIONAIS
## EXERCÍCIO
### Você deverá utilizar o que aprendeu nas aulas de condicionais para desenvolver os desafios a seguir:

1. Seu primeiro desafio é desenvolver um programa para um depósito de bebidas que valide venda de bebidas para maiores de idade (maior ou igual 18 anos) no mercado, o programa deve receber do usuário os valores do nome e ano que ele nasceu e retornar se ele pode comprar bebidas.


In [20]:

from datetime import date
Data= date.today().year

try:
    idade=int(input('Qual a sua idade?\n  '))
    if idade<=17:
        print('Não vendemos bebidas para menores de 18 anos')

    else:
        nome = input('Qual o seu nome?\n  ')
        nascimento = int(input('Qual ano você nasceu?\n  '))
        realIdade = Data-nascimento
        if realIdade <= 17:
            print(f'{nome} a idade que você nos informou e incoerente com sua data de nascimento!\n'
            'Apresente um docomento com foto para validarmos as informações.')
        else:
            print(f'{nome} beba com moderação!')

except:
    print('Erro, leia e responda como está sendo pedido.')



gabriel a idade que você nos informou e incoerente com sua data de nascimento!
Apresente um docomento com foto para validarmos as informações.


### 2. Esse bimestre na faculdade você precisa tirar nota 8 de média para passar na matéria, desenvolva um script que leia a nota de suas últimas 3 provas, tire a média delas e verifique se você passou.


In [24]:
nota1=float(input('Infome a nota da sua primeira prova:  '))
nota2=float(input('Informe a nota da sua segunda prova:  '))
nota3=float(input('Informe a nota da sua segunda prova:  '))

nota=(nota1+nota2+nota3)/3
if nota>=8:
    print (f'Sua nota foi {nota:.2f}, você foi aprovado')

else:
    print (f'Sua nota foi de {nota:.2f}, você ficou abaixo da media')

Sua nota foi 8.53, você foi aprovado


### 3. A fisioterapeuta que você vai gosta muito de brincar com números, e pediu para você criar um sistema que verifique se a altura inserida é par ou impar.


In [32]:
valor = int(input('informe um numero:  '))
soma = valor % 2

if soma == 0:
    print(f'O número {valor} é PAR')
else:
    print(f'O número {valor} é Ímpar')

O número 25 é Ìmpar


### 4. Faça um script que você coloque a quantidade de dias e descubra em qual trimestre esse dia está no ano.


In [47]:
dias =0
try:
    ano=int (input('digite [1] se o ano atual tem 365 dias [2] se o ano tem 366'))
    if ano == 1:
        print ('Ano bissexto')
        print('_'*100 )
        print()

        if dias >=0 or dias <= 366:
            dias=int(input('digite número entre 1 e 366 referente aos dias do ano:\n  '))
                                
            if dias >0 and dias<92:
                print('O dia informado coreesponde ao 1º trimestre: de janeiro a março.')
            elif dias >91 and dias <183:
                print('O dia informado coreespode ao 2º trimestre: de abril a junho.')
            elif dias >182 and dias <274:
                print('O dia informado corresponde ao 3º trimestre: de julho a setembro.')
            else:
                print ('O dia informado correponde ao 4º trimestre: de outubro a dezembro.')
        
        else:    
            print ('o dia desejado esta inconpativel. tente novamente.')


    elif ano == 2:
        print ('O ano e bisiesto')
        print('_'*100 )
        print()

        if dias >=0 or dias <= 368:
            dias=int(input('digite número entre 1 e 366 referente aos dias do ano:\n  '))

            if dias >0 and dias<92:
                print('O dia informado coreesponde ao 1º trimestre: de janeiro a março.')
            elif dias >91 and dias <183:
                print('O dia informado coreespode ao 2º trimestre: de abril a junho.')
            elif dias >182 and dias <275:
                print('O dia informado corresponde ao 3º trimestre: de julho a setembro.')
            else:
                print ('O dia informado correponde ao 4º trimestre: de outubro a dezembro.')
        
        else:
            print ('o dia desejado esta inconpativel. tente novamente.')

except:
    print('Erro')

366
Ano bissexto
____________________________________________________________________________________________________
O dia informado coreespode ao 2º trimestre: de abril a junho.


### 5. Desenvolva um sistema que leia 3 números e diga qual é o maior (se houver empate, exiba os empatados).


In [93]:
a=int(input('Digite o primeiro valor:  '))
b=int(input('Digite o segundo valor:  '))
c=int(input('Digite o terceiro valor:  '))

if a>b and a>c:
    maior=a
elif b>a and b>c:
    maior = b
elif c>a or c>b:
    maior = c

print ('O maior número foi',maior)

if a==b:
    igual = a
    print ('O número',igual, 'se repetiu')
elif b==c:
    igual =b
    print ('O número',igual, 'se repetiu')
else:
    print('Não tivemos números iguais')



O maior número foi 9
Não tivemos números iguais


# Exercício 1
Some os números ímpares consecutivos começando em 1 e terminando 9

In [1]:
comando=range(1,22,2)
soma = 0
for j in comando:
    soma += j
    print('soma dos numeros {}'.format(soma))

soma dos numeros 1
soma dos numeros 4
soma dos numeros 9
soma dos numeros 16
soma dos numeros 25
soma dos numeros 36
soma dos numeros 49
soma dos numeros 64
soma dos numeros 81
soma dos numeros 100
soma dos numeros 121


# Exercício 1
Tire a média de 3 números digitados pelo usuário.

In [12]:
estrutura=range(1,4)
base=0
for i in estrutura:
    a= float(input(f'digite o {i}º numero'))
    base +=a
    print(f'a soma do {i}º número foi {base :.0f}')
media = base / i
print ('A media foi de {:.2f}'.format(media))

a soma do 1º número foi 4
a soma do 2º número foi 13
a soma do 3º número foi 16
A media foi de 5.33


## Exercício 2
1. Pergunte para o usuário de quantos números ele quer tirar a média
2. Tire a média desses números

In [20]:
i=int(input('quantos número você querr tirar a media'))
passo=range(1,i+1)
soma=0
print(f'será {i} periguntas, bora responder e d3scobrir qual a media deles: ')
for a in passo:
    pergunta=float(input(f'Digite o {a}º número:  '))
    soma +=pergunta
    print(f'a soma do {a}º número foi de {soma} ')
media= soma / a
print(f'A media foi de {media :.2f}')

será 5 periguntas, bora responder e d3scobrir qual a media deles: 
a soma do 1º número foi de 1.0 
a soma do 2º número foi de 22.0 
a soma do 3º número foi de 31.0 
a soma do 4º número foi de 38.0 
a soma do 5º número foi de 72.5 
A media foi de 14.50


## Exercício 1
Some números digitados pelo usuário até a soma deles ultrapassar 20

In [31]:
soma = 0
while soma < 20:
    s=int(input('digite um número:  '))
    soma += s
print (soma)

22


# Exercicio
Faça um programa que caucule a tabuada do numero digitada pelo usuário.

In [87]:
num=int(input('qual tabuada você quer: '))

for i in range(1,11):
    
    print('{} * {} = {}' .format(num, i, num*i))

7 * 1 = 7
7 * 2 = 14
7 * 3 = 21
7 * 4 = 28
7 * 5 = 35
7 * 6 = 42
7 * 7 = 49
7 * 8 = 56
7 * 9 = 63
7 * 10 = 70


# Exercício 2
Faça um programa que o usuário irá digitar 10 números e você irá mostrar:

1. Qual foi o menor valor
2. Qual foi o maior valor
3. Qual a média dos valores

In [112]:
i=[]
for a in range(1,11):
    perguntas=int(input(f'Digite o {a}º número: '))
    i.append(perguntas)

print('O maior número foi:',max(i))
print(f'O menor numero foi: {min(i)}')
print('A media foi: {:.2f}'.format (sum(i)/a))

O maior número foi: 652
O menor numero foi: 2
A media foi: 135.90


# Exercício 3
Faça um programa que calcula a Sequência de Fibonacci de N números, onde N é dado pelo usuário.

Essa sequência é construída começando com os números 0 e 1. O próximo número da sequência é sempre a soma dos 2 anteriores.

Sequência: 0 - 1 - 1 - 2 - 3 - 5 - 8 - 13 - 21 - ...

In [105]:
valor1=0
valor2=1

n=int(input('Informe o numero para a sequencia de Fibonacci: '))
print(f'valos execultar {n} primeriros termos da sequencia de Fibonacci.')

print(valor1, '-' ,valor2, end='')

for a in range(3,n):
    valor3 = valor1+valor2
    print(f' - {valor3}', end='')
    valor1 = valor2
    valor2 = valor3
    contador += 1



valos execultar 30 primeriros termos da sequencia de Fibonacci.
0 - 1 - 1 - 2 - 3 - 5 - 8 - 13 - 21 - 34 - 55 - 89 - 144 - 233 - 377 - 610 - 987 - 1597 - 2584 - 4181 - 6765 - 10946 - 17711 - 28657 - 46368 - 75025 - 121393 - 196418 - 317811

# Exercício 1
Dada a tupla abaixo:

t1 = ('Doce', 'Churrasco', 'Bala ', 'Tapioca', 'Pizza ', 'Feijão ', 
      'Arroz ', 'Açaí ', 'Paçoca ', 'Acarajé', 'Pamonha', 'Dobradinha', 'Rapadura')     
#### Faça um programa que:    

O usuário digita uma comida e seu programa diz em qual posição do cardápio esse item está      
Printe os itens em ordem alfabética para o usuário

In [1]:
try:
      t1 = ('Doce', 'Churrasco', 'Bala', 'Tapioca', 'Pizza', 'Feijão', 
      'Arroz', 'Açaí', 'Paçoca', 'Acarajé', 'Pamonha', 'Dobradinha', 'Rapadura')

      print('-'*5, 'Cardápio', '-'*5)
      for i in t1:
            print(f'Temos: {i:.>13}')
      print('-'*50)
      a=input('Oque deseja? ').capitalize()
      print(f'A opção escolhida foi {a} e esta na posição:',t1.index(a))


except:
      print('Erro de', ValueError)

print('-'*50,'\n',
      ' \n',
      ' \n',
      sorted(t1))


----- Cardápio -----
Temos: .........Doce
Temos: ....Churrasco
Temos: .........Bala
Temos: ......Tapioca
Temos: ........Pizza
Temos: .......Feijão
Temos: ........Arroz
Temos: .........Açaí
Temos: .......Paçoca
Temos: ......Acarajé
Temos: ......Pamonha
Temos: ...Dobradinha
Temos: .....Rapadura
--------------------------------------------------
Erro de <class 'ValueError'>
-------------------------------------------------- 
  
  
 ['Acarajé', 'Arroz', 'Açaí', 'Bala', 'Churrasco', 'Dobradinha', 'Doce', 'Feijão', 'Pamonha', 'Paçoca', 'Pizza', 'Rapadura', 'Tapioca']


# Exercício 2
Dado a tupla abaixo:

t1 = ('Doce', 2.3, 'Bala ', 0.15, 'Pizza ', 28.9, 'Arroz ', 4.5, 'Paçoca ', 0.5, 'Pamonha', 5.4)     
Os dados estão organizados de forma que tem o valor do produto e logo após o seu preço. Faça um programa que:

Mostre o cardápio de forma organizada     
Resultado esperado:

In [1]:
t1 = ('Doce', 2.3, 'Bala ', 0.15, 'Pizza ', 28.9, 'Arroz ', 4.5, 'Paçoca ', 0.5, 'Pamonha', 5.4)
car=' Cardapio '
print('','-'*32,'\n',
'{:-^32}'.format(car),'\n',
'-'*32)
a=-2
b=-1

for i in range(1,int((len(t1)+2)/2)):
    a+=2
    b+=2
    print(f' {t1[a]:.<23} R$ {t1[b]: >5}')


 -------------------------------- 
 ----------- Cardapio ----------- 
 --------------------------------
 Doce................... R$   2.3
 Bala .................. R$  0.15
 Pizza ................. R$  28.9
 Arroz ................. R$   4.5
 Paçoca ................ R$   0.5
 Pamonha................ R$   5.4


# Exercício 3
Dado a mesma tupla do exercício anterior faça um programa que:     

O usuário visualiza o cardápio e digita qual item ele quer comprar. A seguir pergunta se ele quer algo a mais. Caso ele queira vai somando todos os valores do pedido numa variável soma.     
No final printe para o usuário o valor total da conta dele

In [ ]:
# cores e estilos
# padrão par ao python   \033[1;31;47m
# 0 normal
# 1 negrito
# 4 sublinhado
# 7 fundo
# *****     COR DA FONTE    *****
# 30 cor branca
# 31 cor vermelha
# 32 cor verde
# 33 cor amarela
# 34 cor azul 
# 35 cor roxo
# 36 cor verde água
# 37 cor cinza
# *****     COR DO FUNDO    ******
# 40 cor do fundo branco
# 41 cor do fundo vermelho
# 42 cor do fundo verde
# 43 cor do fundo amarelo
# 44 cor do fundo azul
# 45 cor do fundo roxo
# 46 cor do fundo verde agua
# 47 cor do funfo cinza 

t1 = ('Doce', 2.3, 'Bala', 0.15, 'Pizza', 28.9, 'Arroz', 4.5, 'Paçoca', 0.5, 'Pamonha', 5.4)
car=' Cardapio '
print('','-'*32,'\n','{:-^32}'.format(car),'\n','-'*32)
a=-2
b=-1

for i in range(1,int((len(t1)+2)/2)):    
    a+=2
    b+=2
    print(f' {t1[a]:.<23} R$ {t1[b]: >5}')
        
print('_'*50,'\n')
valor_pedido=0
comanda=[]
while True:
    try:        
        pedido=str(input('Oque deseja? ')).capitalize()
        comanda.append(pedido)
        valor_pedido+=(t1[t1.index(pedido)+1])
        pergunta=input('Algo mais?\n' 'Pressione \033[1;37;44m"S"\033[0m para continuar:  ')[0].lower
    
        while pergunta != 's':
            
            print('_'*50,' \n','Os itens escolhidos foram:\n')

            for i,c  in enumerate(comanda):
                print(f'{c}' )
            print(f'\nvalor total do pedido foi:\nR$ {valor_pedido:.2f}')
            break

    except ValueError:
        loop = input('\033[1;31;40mAlgo deu errado!\n'
'\033[0;mDeseja tentar novamente? (S/N)\n')[0].lower()
        if loop == 's':
            continue
        else:
            print ('Fim do programa, até logo')

 -------------------------------- 
 ----------- Cardapio ----------- 
 --------------------------------
 Doce................... R$   2.3
 Bala................... R$  0.15
 Pizza.................. R$  28.9
 Arroz.................. R$   4.5
 Paçoca................. R$   0.5
 Pamonha................ R$   5.4
__________________________________________________ 

Fim do programa, até logo
Fim do programa, até logo
